mobile-app/app/screens/AppNavigator/screens/Loans/screens/CreateVaultScreen.tsx

```diff
@@ -1,41 +1,82 @@
 import { View } from '@components'
 import { Button } from '@components/Button'
-import { ThemedIcon, ThemedScrollView, ThemedSectionTitle, ThemedText, ThemedTouchableOpacity, ThemedView } from '@components/themed'
+import {
+  ThemedIcon,
+  ThemedScrollView,
+  ThemedSectionTitle,
+  ThemedText,
+  ThemedTouchableOpacity,
+  ThemedView
+} from '@components/themed'
 import { StackScreenProps } from '@react-navigation/stack'
 import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
 import { tailwind } from '@tailwind'
 import { translate } from '@translations'
-import React, { useEffect, useState } from 'react'
+import React, { Dispatch, useEffect, useState } from 'react'
 import NumberFormat from 'react-number-format'
 import { LoanParamList } from '../LoansNavigator'
 import { LoanScheme } from '@defichain/whale-api-client/dist/api/loan'
 import BigNumber from 'bignumber.js'
-import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { NativeLoggingProps, useLogger } from '@shared-contexts/NativeLoggingProvider'
 import { useDispatch, useSelector } from 'react-redux'
 import { fetchLoanSchemes } from '@store/loans'
 import { RootState } from '@store'
+import { hasTxQueued, transactionQueue } from '@store/transaction_queue'
+import { hasTxQueued as hasBroadcastQueued } from '@store/ocean'
+import { DFITokenSelector, DFIUtxoSelector } from '@store/wallet'
+import { ConversionMode, dfiConversionCrafter } from '@api/transaction/dfi_converter'
 
 type Props = StackScreenProps<LoanParamList, 'CreateVaultScreen'>
 
-export function CreateVaultScreen ({ navigation, route }: Props): JSX.Element {
+export function CreateVaultScreen ({
+  navigation,
+  route
+}: Props): JSX.Element {
   const dispatch = useDispatch()
   const client = useWhaleApiClient()
   const loanSchemes: LoanScheme[] = useSelector((state: RootState) => state.loans.loanSchemes)
   const logger = useLogger()
   const [fee, setFee] = useState<BigNumber>(new BigNumber(0.0001))
   const [selectedLoanScheme, setSelectedLoanScheme] = useState<LoanScheme | undefined>(route.params?.loanScheme)
-  const onSubmit = (): void => {
-    if (selectedLoanScheme === undefined) {
+  const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
+  const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const DFIUtxo = useSelector((state: RootState) => DFIUtxoSelector(state.wallet))
+  const DFIToken = useSelector((state: RootState) => DFITokenSelector(state.wallet))
+  const isConversionRequired = new BigNumber(2).gt(DFIUtxo.amount)
```

Comment:
I think we also need to check if `DFIToken` has enough to convert?

Reply:
It will throw an error if it doesn't have DFIToken and DFIUTXO on submit. So should be ok